src/hotspot/share/opto/movenode.hpp

```diff
@@ -87,14 +87,14 @@ class CMoveLNode : public CMoveNode {
 //------------------------------CMovePNode-------------------------------------
 class CMovePNode : public CMoveNode {
   public:
-  CMovePNode( Node *c, Node *bol, Node *left, Node *right, const TypePtr* t ) : CMoveNode(bol,left,right,t) { init_req(Control,c); }
+  CMovePNode(Node* bol, Node* left, Node* right, const TypePtr* t) : CMoveNode(bol, left, right, t) {}
```

Comment:
I looked on history and this control setting was added at the very beginning.
May be it was done to avoid bypassing safepoints where oops could be modified (objects moved) and result of `CMoveP` before and after safepoint will be different.
I would be careful about changing it and `CMoveN`.

Reply:
I have thought about it and can't think of a reason for the control input. 
Firstly, the oop map is created after scheduling and should take care of this seemlessly.
Secondly, the control input does not prevent a CMoveP from moving down past a safepoint. And since these nodes have depends_only_on_test == true, they can be moved from a test to an equivalent dominating test, skipping any safepoint in between, which means these nodes can be moved up past a safepoint, too.
As a result, I don't think the control input of CMoveP and CMoveN is necessary and can be safely removed.

Reply:
Any undesireable performance effects if CMove node floats up beyond its (original) control? It may have changed on modern hardware, but my recollection is cmov instructions were quite expensive to execute.

Reply:
@iwanowww A cmov is as cheap as an add so I don't think there will be any performance effect. I believe what you mean by cmov being expensive is that it has 1 cycle of latency on both of its inputs, while a well-predicted if diamond has 0 cycle of latency on the chosen input only.

Reply:
Right, my memory failed me. Indeed I was referring to *relative* slowness of cmov instructions compared to well-predicted conditional jumps which is not relevant here.

Reply:
> As a result, I don't think the control input of CMoveP and CMoveN is necessary and can be safely removed.

Okay. Let test it in JDK 25 and see how it goes. We have time to fix it if we find something wrong.
@dafedafe ran tier1-5 our testing and I don't see any new failures.